### PR TITLE
Fix topology crash due to selfLink deprecation

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
**Fixes**: 
https://issues.redhat.com/browse/ODC-5563

**Analysis / Root cause**: 
https://github.com/kubernetes/enhancements/issues/1164 removed the `selfLink` attribute. This attribute was being used to determine the managed by resource for the operator group.

**Solution Description**: 
Remove the dependency on `selfLink` by passing back the name of the CSV and keeping this name as part of the data for the operator backed service group node.

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

/kind bug

### CODE REVIEW

**Review by rottencandy — APPROVED** [review]
/lgtm

**Comment by rottencandy** [comment]
@jeff-phillips-18 looks like changing `getOperatorGroupResources` has broken some test cases.

**Comment by jeff-phillips-18** [comment]
> @jeff-phillips-18 looks like changing `getOperatorGroupResources` has broken some test cases.

Thanks. Fixing.

**Comment by jeff-phillips-18** [comment]
> @jeff-phillips-18 looks like changing `getOperatorGroupResources` has broken some test cases.

Fixed. Thanks.

**Comment by invincibleJai** [comment]
/approve

Verified the changes , works as expected

**Comment by spadgett** [comment]
Holding until #8289 merges to unblock CI.

/hold

**Pull request merged**
